temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflowContext.java

```diff
@@ -133,6 +137,24 @@ public void setHeadInterceptor(WorkflowOutboundCallsInterceptor head) {
     }
   }
 
+  @Override
+  public void setDefaultActivityOptions(ActivityOptions defaultActivityOptions) {
+    this.defaultActivityOptions = defaultActivityOptions;
+  }
```

Comment:
I would remove setters and initialize the activityOptions fields in the constructor. 

Note that SyncWorkflow already receives implementationOptions, so it can pass the activity options to the SyncWorkflowContext on construction in the SyncWorkflow.start method.